src/fields.jl

```diff
@@ -52,3 +52,231 @@ Field(LX, data, grid::Grid1D) = Field1D(LX, data, grid)
 Constructs a 2D field of `data` at location `(LX, LY)` on `grid`.
 """
 Field(LX, LY, data, grid::Grid2D) = Field2D(LX, LY, data, grid)
+
+############################################################################
+#-------------------------------Interpolation------------------------------#
+
+𝐼xᶠ(i::Int, f::Field1D{Face})   = (f.data[i] + f.data[i+1]) / 2
+𝐼xᶜ(i::Int, f::Field1D{Centre}) = (f.data[i] + f.data[i-1]) / 2
+
+𝐼xᶠᶜ(i::Int, j::Int, f::Field2D{Face, Centre})   = (f.data[i, j] + f.data[i+1, j]) / 2
+𝐼xᶜᶜ(i::Int, j::Int, f::Field2D{Centre, Centre}) = (f.data[i, j] + f.data[i-1, j]) / 2
+𝐼yᶜᶠ(i::Int, j::Int, f::Field2D{Centre, Face})   = (f.data[i, j+1] + f.data[i, j]) / 2
+𝐼yᶜᶜ(i::Int, j::Int, f::Field2D{Centre, Centre}) = (f.data[i, j] + f.data[i, j-1]) / 2
+
+"""
+    𝐼x!(output::Field1D{Centre}, input::Field1D{Face})
+
+Interpolates a 1D field of `data` from face to centre in x-direction.
```

Comment:
```suggestion
Interpolates a 1D `input` field that lives in `Face`s to `output` field that lives on `Centre`s.
```

I'm resuggesting. :)
Also this suggestion applies to all docstrings. I mean, the docstring should be informative of what the actual function does and it should describe a bit how the functions arguments are involved. The way is written now the docstring does not explain that there are two fields as arguments and the function interpolates the latter to the former. It makes a vague statement about "`data` from face to centre". Could the user understand what this function is doing without looking at the source code?

Reply:
Yeah. This is more readable.